batch/sql/add-batch-updates.sql

```diff
@@ -0,0 +1,446 @@
+START TRANSACTION;
+
+# Foreign key constraint for batch_id needs to be added in a subsequent PR in order for the migration to work
+CREATE TABLE IF NOT EXISTS `batch_updates` (
+  `batch_id` BIGINT NOT NULL,
+  `update_id` VARCHAR(40) NOT NULL,
+  `start_job_id` INT,
+  `n_jobs` INT NOT NULL,
+  `committed` BOOLEAN NOT NULL DEFAULT FALSE,
+  `time_created` BIGINT,
+  `time_committed` BIGINT,
+  PRIMARY KEY (`batch_id`, `update_id`)
+) ENGINE = InnoDB;
+CREATE INDEX `batch_updates_committed` ON `batch_updates` (`batch_id`, `committed`);
+CREATE INDEX `batch_updates_start_job_id` ON `batch_updates` (`batch_id`, `start_job_id`);
+
+ALTER TABLE batches ADD COLUMN update_added BOOLEAN DEFAULT FALSE, ALGORITHM=INSTANT;
+ALTER TABLE batches_inst_coll_staging ADD COLUMN update_id VARCHAR(40), ALGORITHM=INSTANT;
+ALTER TABLE batch_inst_coll_cancellable_resources ADD COLUMN update_id VARCHAR(40), ALGORITHM=INSTANT;
+
+DELIMITER $$
+
+DROP TRIGGER IF EXISTS batches_before_insert $$
+CREATE TRIGGER batches_before_insert BEFORE INSERT ON batches
+FOR EACH ROW
+BEGIN
+  DECLARE cur_update_id VARCHAR(40);
+  DECLARE new_update_id VARCHAR(40);
+
+  SELECT update_id INTO cur_update_id FROM batch_updates WHERE batch_id = NEW.id;
+
+  IF cur_update_id IS NULL THEN
+    SET new_update_id = LEFT(MD5(RAND()), 8);
+
+    INSERT INTO `batch_updates` (`batch_id`, `update_id`, start_job_id, n_jobs, `committed`, time_created, time_committed)
+      VALUES (NEW.id, new_update_id, 1, NEW.n_jobs, NEW.state != 'open', NEW.time_created, NEW.time_closed)
+    ON DUPLICATE KEY UPDATE n_jobs = n_jobs;
+  END IF;
+
+  SET NEW.update_added = TRUE;
+END $$
+
+DROP TRIGGER IF EXISTS batches_before_update $$
+CREATE TRIGGER batches_before_update BEFORE UPDATE ON batches
```

Comment:
This is missing from estimated-current, but is correct. Since this is getting deleted in the next PR I wouldn't bother.